src/content/docs/apm/agents/java-agent/api-guides/ai-monitoring.mdx

```diff
@@ -0,0 +1,146 @@
+
+1. Implement the `LlmTokenCountCallback` so it overrides the `calculateLlmTokenCount(String model, String content)` method. This calculates a token count based on a given LLM model name and the LLM message content or prompt: 
+
+    ```java
+        class MyTokenCountCallback implements LlmTokenCountCallback {
+            @Override
+            public int calculateLlmTokenCount(String model, String content) {
+                // Implement custom token calculating logic here based on the LLM model and content.
+                // Return an integer representing the calculated token count.
+                return 0;
+            }
+        }
+    ```
+1.  Create an instance of the `LlmTokenCountCallback` implementation to register the callback, then pass it to the `setLlmTokenCountCallback` API. For example: 
+
+    ```java
+        LlmTokenCountCallback myTokenCountCallback = new MyTokenCountCallback();
+            // The callback needs to be registered at some point before invoking the LLM model
+        NewRelic.getAgent().getAiMonitoring.setLlmTokenCountCallback(myTokenCountCallback);
+    ```
+
+To use the callback, implement `LlmTokenCountCallback` so that:
+
+* It returns an integer that represents the number of tokens for a particular prompt, completion message, or embedding. 
+* If the values are less than or equal to 0, `LlmTokenCountCallbacks` will not be attached to an event.
+
+Keep in mind that this API should be called only once. Calling this API multiple times will replace each previous callback. 
+
+## Record user feedback [#user-feedback]
+
+AI monitoring can correlate trace IDs between a generated message from your LLM models and an end user's feedback. The [`recordLlmFeedbackEvent`](https://newrelic.github.io/java-agent-api/javadoc/com/newrelic/api/agent/AiMonitoring.html#recordLlmFeedbackEvent) API constructs an argument with a map of the `LlmFeedbackEventAttributes.Builder` class. 
+
+Recording user feedback takes two steps:
+
+1. Use the [`TraceMetadata.getTraceId()`](https://newrelic.github.io/java-agent-api/javadoc/com/newrelic/api/agent/TraceMetadata.html#getTraceId) API to acquire the trace ID for the currently executing transaction:
+
+    ```java
+    String traceId = NewRelic.getAgent().getTraceMetadata().getTraceId();
+    ```
+
+1. Add the [`recordLlmFeedbackEvent(Map<String, Object> llmFeedbackEventAttributes)`](https://newrelic.github.io/java-agent-api/javadoc/com/newrelic/api/agent/AiMonitoring.html#recordLlmFeedbackEvent) to correlate the trace ID with a feedback event. Here is an example of how you might record an LLM feedback event:
```

Comment:
Should this be step `2.` or does the rendering magically change that?

Reply:
Yes! markdown fixes ordered lists automagically when they all have 1.s Confirmed it looks good in the preview, too